src/main/java/core/basesyntax/service/FileServiceImpl.java

```
@@ -0,0 +1,42 @@
+package core.basesyntax.service;
+
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class FileServiceImpl implements FileService {
+    private static final String FILE_PATH = "src/main/resources/fruits.csv";
+    private static final String REPORT_FILE_PATH = "src/main/resources/report_fruits.csv";
+
+    @Override
+    public List<String> readFromFile() {
+        List<String> stringList = new ArrayList<>();
+
+        try (BufferedReader bufferedReader = new BufferedReader(new FileReader(FILE_PATH));) {
+            String line;
+
+            while ((line = bufferedReader.readLine()) != null) {
+                stringList.add(line);
+            }
+
+        } catch (IOException e) {
+            throw new RuntimeException("Can't find file by path: " + e);
```

This violates checklist item #6: 'Don't ignore exceptions in try-catch blocks'. When rethrowing an exception, it's important to pass the original exception as the `cause`. The correct way is to use the `RuntimeException(String message, Throwable cause)` constructor, like this: `throw new RuntimeException("Can't find file by path: " + FILE_PATH, e);`. This preserves the original stack trace for better debugging.